Running layout tests with Site Isolation enabled, spawns many WebContent processes

pvollan opened the pull request:
#### 35541d36555ccac06542e67295901757d2a47e1e
<pre>
Running layout tests with Site Isolation enabled, spawns many WebContent processes
<a href="https://bugs.webkit.org/show_bug.cgi?id=298779">https://bugs.webkit.org/show_bug.cgi?id=298779</a>
<a href="https://rdar.apple.com/160474658">rdar://160474658</a>

Reviewed by Sihui Liu.

Prewarmed WebContent processes are oftentimes unused when running layout tests. Disabling prewarming reduces the number of
WebContent processes launched in a single test.

* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::generatePageConfiguration):

Canonical link: <a href="https://commits.webkit.org/299956@main">https://commits.webkit.org/299956@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a77929e28459e03802b60042329c47a940a2cace

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120684 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40378 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31030 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127077 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72759 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4f0cdd2f-d0b8-4854-a8ec-a0c85f70dfd4) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122560 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41075 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48955 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91664 "11 flakes 57 failures") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60921 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c589bf36-cf6c-4380-b917-c3874dbcf422) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123636 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32800 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108178 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72214 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fb68a3cf-9aac-43b6-b48a-6326e10b7542) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31828 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26285 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70682 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102287 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26467 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129945 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47605 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36148 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100285 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47973 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104360 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100124 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45573 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23602 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44265 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19175 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47467 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46936 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50282 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48622 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -973,6 +973,7 @@ WKRetainPtr<WKPageConfigurationRef> TestController::generatePageConfiguration(co
 {
     if (!m_context || !m_mainWebView || !m_mainWebView->viewSupportsOptions(options)) {
         auto contextConfiguration = generateContextConfiguration(options);
+        WKContextConfigurationSetPrewarmsProcessesAutomatically(contextConfiguration.get(), false);
```
Should we just do this when `options.runInCrossOriginFrame()` and `options.siteIsolationEnabled()` are true, if this is only a concern for running with --site-isolation?

pvollan replied:
That is a good point. I have updated the patch.

Thanks for reviewing!

pvollan committed:
Running layout tests with Site Isolation enabled, spawns many WebContent processes
https://bugs.webkit.org/show_bug.cgi?id=298779
rdar://160474658

Reviewed by Sihui Liu.

Prewarmed WebContent processes are oftentimes unused when running layout tests. Disabling prewarming reduces the number of
WebContent processes launched in a single test.

* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::generatePageConfiguration):

Canonical link: https://commits.webkit.org/299956@main
